packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -182,14 +196,19 @@ contract Bridge is EssentialContract, IBridge {
 
         if (messageStatus[msgHash] != Status.NEW) revert B_STATUS_MISMATCH();
 
-        bool received = _proveSignalReceived(msgHash, message.srcChainId, proof);
-        if (!received) revert B_NOT_RECEIVED();
+        ISignalService signalService = ISignalService(resolve("signal_service", false));
+        if (!_proveSignalReceived(signalService, msgHash, message.srcChainId, proof)) {
+            revert B_NOT_RECEIVED();
+        }
 
         Status status;
         uint256 refundAmount;
 
         // Process message differently based on the target address
-        if (message.to == address(0) || message.to == address(this)) {
+        if (
+            message.to == address(0) || message.to == address(this)
+                || message.to == address(signalService)
+        ) {
```

Comment:
Anything who is CrossChainOwned() is also dangerous to call, no ? 🤔 
<img width="1105" alt="kép" src="https://github.com/taikoxyz/taiko-mono/assets/51912515/e4fe6d5d-ebaa-41cc-9e1c-3e5550dab818">

Reply:
OK, i think it is fine for TaikoL2 (which is `CrossChainOwned`), because owner still has to be the `ctx.from`.